Modules/_multiprocessing/semaphore.c

```diff
@@ -511,7 +512,7 @@ semlock_rebuild(PyTypeObject *type, PyObject *args)
         strcpy(name_copy, name);
     }
 
-#ifndef MS_WINDOWS
+#if !defined(MS_WINDOWS) && !defined(USE_UNNAMED_SEMAPHORES) // if we're using unnamed, then we got it via fork, can't sem_open
```

Comment:
```suggestion
#ifndef USE_UNNAMED_SEMAPHORES
```
Define USE_UNNAMED_SEMAPHORES in the `#ifdef MS_WINDOWS` block above and this simplifies the check here.